core/src/main/scala/kafka/server/ConfigHandler.scala

```diff
@@ -72,21 +102,49 @@ object ClientConfigOverride {
 }
 
 /**
- * The ClientIdConfigHandler will process clientId config changes in ZK.
- * The callback provides the clientId and the full properties set read from ZK.
- * This implementation reports the overrides to the respective ClientQuotaManager objects
- */
-class ClientIdConfigHandler(private val quotaManagers: Map[Short, ClientQuotaManager]) extends ConfigHandler {
-
-  def processConfigChanges(clientId: String, clientConfig: Properties) = {
+  * The ClientIdConfigHandler will process clientId config changes in ZK.
+  * The callback provides the clientId and the full properties set read from ZK.
+  * This implementation reports the overrides to the respective ClientQuotaManager objects
+  */
+class ClientIdConfigHandler(private val quotaManagers: QuotaManagers) extends ConfigHandler {
+  def processConfigChanges(clientId: String, clientConfig: Properties) {
     if (clientConfig.containsKey(ClientConfigOverride.ProducerOverride)) {
-      quotaManagers(ApiKeys.PRODUCE.id).updateQuota(clientId,
+      quotaManagers.produce.updateQuota(clientId,
         new Quota(clientConfig.getProperty(ClientConfigOverride.ProducerOverride).toLong, true))
     }
 
     if (clientConfig.containsKey(ClientConfigOverride.ConsumerOverride)) {
-      quotaManagers(ApiKeys.FETCH.id).updateQuota(clientId,
+      quotaManagers.fetch.updateQuota(clientId,
         new Quota(clientConfig.getProperty(ClientConfigOverride.ConsumerOverride).toLong, true))
     }
   }
 }
+
+/**
+  * The BrokerConfigHandler will process individual broker config changes in ZK.
+  * The callback provides the brokerId and the full properties set read from ZK.
+  * This implementation reports the overrides to the respective ReplicationQuotaManager objects
+  */
+class BrokerConfigHandler(private val brokerConfig: KafkaConfig, private val quotaManagers: QuotaManagers) extends ConfigHandler with Logging {
+  def processConfigChanges(brokerId: String, properties: Properties) {
+    if (brokerConfig.brokerId == brokerId.trim.toInt) {
+      val limit = if (properties.containsKey(ThrottledReplicationRateLimitProp)) properties.getProperty(ThrottledReplicationRateLimitProp).toLong else Defaults.ThrottledReplicationLimit
```

Comment:
Yes, we are doing similar things in KIP-55. We are deprecating the static client quota configs in the broker in favor of dynamic quotas. So, for new broker configs, if it can be made dynamically, it seems it's less confusing to also add a static config in the broker property.